ua.rakuguide.adoc

```diff
@@ -0,0 +1,1058 @@
+@animals.push("сова");
+say "Тепер у зоопарку є: " ~ @animals;
+say "Перша тварина, яка в нас з'явилася, це " ~ @animals[0];
+@animals.pop;
+say "На жаль сова втекла та в нас лишилися: " ~ @animals;
+say "Ми закриваємо зоопарк та залишаемо собі лише одну тварину";
+say "Ми плануємо віддати: " ~ @animals.splice(1,2) ~ " та залишити " ~ @animals;
+----
+
+.`Вивід`
+----
+У зоопарку є 3 тварини
+Ці тварини: верблюд вікунья лама
+Я маю намір віддати до зоопарку сову
+Тепер у зоопарку є: верблюд вікунья лама сова
+Перша тварина, яка в нас з'явилася, це верблюд
+На жаль сова втекла та в нас лишилися: верблюд вікунья лама
+Ми закриваємо зоопарк та залишаємо собі лише одну тварину
+Ми плануємо віддати: вікунья лама та залишити верблюд
+----
+
+.Пояснення
+`.elems` повертає кількість елементів масиву. +
+`.push()` додає один чи більше елементів в кінець масиву. +
+Ми можемо звернутися до певного елемента масиву вказавши його положення `@animals[0]`. +
+`.pop` видаляє останній елемент масиву та повертає його. +
+`.splice(a,b)` видалить (та поверне) `b` елементів починаючи з позиції `a`.
+
+==== Масиви фіксованого розміру
+
+Оголошення звичайного масиву виглядає наступним чином:
+
+[source,perl6]
+my @array;
+
+Звичайний масив может бути довільного розміру, тому його називають автоматично розширюваним. +
+Такий масив приймає довільну кількість значень без будь-яких обмежень.
+
+Також, на противагу, ми можемо створити масив фіксованого розміру. +
+Доступ до таких масивів є неможливим за межами їх визначеного розміру.
+
+[source,perl6]
+my @array[3];
+
+Цей масив здатен зберігати завбільшки 3 значення, з індексами від 0 до 2.
+
+[source,perl6]
+----
+my @array[3];
+@array[0] = "перше значення";
+@array[1] = "друге значення";
+@array[2] = "третє значення";
+----
+
+Ви не зможете додати четверте значення до такого масиву.
+
+[source,perl6]
+----
+my @array[3];
+@array[0] = "перше значення";
+@array[1] = "друге значення";
+@array[2] = "третє значення";
+@array[3] = "четверте значення";
+----
+
+----
+Index 3 for dimension 1 out of range (must be 0..2)
+----
+
+==== Багатовимірні масиви
+Масиви, які ми бачили до цього часу є одновимірними. +
+На щастя, у Raku ми можемо визначати багатовимірні масиви.
+
+[source,perl6]
+my @tbl[3;2];
+
+Цей масив двовимірний.
+Перший вимір може збарігати завбільшки 3 значення, та другий вимір не більше ніж 2 значення.
+
+Думвйте про це, як про таблицю 3х2.
+
+[source,perl6]
+----
+my @tbl[3;2];
+@tbl[0;0] = 1;
+@tbl[0;1] = "x";
+@tbl[1;0] = 2;
+@tbl[1;1] = "y";
+@tbl[2;0] = 3;
+@tbl[2;1] = "z";
+say @tbl
+----
+
+----
+[[1 x] [2 y] [3 z]]
+----
+
+.Візуальна репрезентація масива:
+----
+[1 x]
+[2 y]
+[3 z]
+----
+
+NOTE: Повну інструкцію до масивів можна знайти за посиланням: https://docs.raku.org/type/Array
+
+==== Хеши (асоціативні масиви)
+[source,perl6]
+.Хеш - це набір пар Ключ/Значення.
+----
+my %capitals = ('UK','Лондон','Ukraine','Київ');
+say %capitals;
+----
+
+[source,perl6]
+.Інший стислий спосіб заповнення хеша:
```

Comment:
It needs a comma I'd say?

Reply:
not really 
```Інший - стислий - спосіб```  to emphasize the compactness